### PR TITLE
store: filter on map id prefix/suffix

### PR DESCRIPTION
**Pull request by bejito**
Update the store to allow for prefix and suffix filtering on map ID.
- Postgres store should work pretty ok: `LIKE %` is a bit more efficient than a regexp lookup. I updated the index on map ID to add the pattern lookup opclass.
- For ES, the prefix filter is very fact, but the suffix filter is not more efficient than a regex lookup. One solution could be to add the reversed mapID to the mapping and do a prefix filter on it.
- not sure about the rethinkDB part, but it passes the tests 🤞

---
This change is [Reviewable](https://reviewable.io/reviews/stratumn/go-indigocore/428)

### CODE REVIEW

**Comment by bejito** [comment]
___
*[store/store_test.go, line 129 at r1](https://reviewable.io/reviews/stratumn/go-indigocore/428#-LH7RYau51n_GrZPPXDD-r1-129:-LH7Tlkc4ofC-JfVtRBL:bnwjr60) ([raw file](https://github.com/stratumn/go-indigocore/blob/f2d975b8e1e02806ff2299b7041d12aa1a9fd02f/store/store_test.go#L129)):*
<details><summary><i>Previously, golangci wrote…</i></summary><blockquote>

github.com/stratumn/go-indigocore/store_test.args composite literal uses unkeyed fields
</blockquote></details>

why u lik dis golangci

**Comment by bejito** [comment]
Ok don't move, I'll probably pass the filter on the `GetMapIDs` rather

File: store/store_test.go [hunk 1]
```diff
@@ -249,12 +251,38 @@ func TestMapFilter_Match(t *testing.T) {
 			args:   args{defaultTestingSegment()},
 			want:   false,
 		},
+		{
+			name:   "Good prefix",
+			fields: fields{Prefix: "TheMap"},
+			args:   args{defaultTestingSegment()},
```
Comment (golangci) [inline store/store_test.go]: github.com/stratumn/go-indigocore/store_test.args composite literal uses unkeyed fields

File: store/store_test.go [hunk 2]
```diff
@@ -249,12 +251,38 @@ func TestMapFilter_Match(t *testing.T) {
...
+			want:   true,
+		},
+		{
+			name:   "Bad prefix",
+			fields: fields{Prefix: "TheMob"},
+			args:   args{defaultTestingSegment()},
```
Comment (golangci) [inline store/store_test.go]: github.com/stratumn/go-indigocore/store_test.args composite literal uses unkeyed fields

File: store/store_test.go [hunk 3]
```diff
@@ -249,12 +251,38 @@ func TestMapFilter_Match(t *testing.T) {
...
+			want:   false,
+		},
+		{
+			name:   "Good suffix",
+			fields: fields{Suffix: "MapId"},
+			args:   args{defaultTestingSegment()},
```
Comment (golangci) [inline store/store_test.go]: github.com/stratumn/go-indigocore/store_test.args composite literal uses unkeyed fields

**Comment by bejito** [comment]
___
*[rethinkstore/rethinkstore.go, line 320 at r1](https://reviewable.io/reviews/stratumn/go-indigocore/428#-LH7mg_z8njImvHS9VfX:-LH8LfjfCv0vLYvO3o7Y:b-896fix) ([raw file](https://github.com/stratumn/go-indigocore/blob/f2d975b8e1e02806ff2299b7041d12aa1a9fd02f/rethinkstore/rethinkstore.go#L320)):*
<details><summary><i>Previously, alexppxela (Alexandre Thibault) wrote…</i></summary><blockquote>

comment
</blockquote></details>

Done.

**Comment by bejito** [comment]
___
*[store/storetestcases/findsegments.go, line 404 at r1](https://reviewable.io/reviews/stratumn/go-indigocore/428#-LH8PNpU3M-i5tShEPVI:-LH8h3aT4N8e5eHmwVlB:b-bgecue) ([raw file](https://github.com/stratumn/go-indigocore/blob/f2d975b8e1e02806ff2299b7041d12aa1a9fd02f/store/storetestcases/findsegments.go#L404)):*
<details><summary><i>Previously, t-bast (Bastien Teinturier) wrote…</i></summary><blockquote>

I would keep these tests though, it feel safer to also test FindSegment's filtering
</blockquote></details>

@t-bast I actually did not put the map ID prefix/suffix filtering on FindSegments... Do you think I should ?

**Pull request merged**